source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -179,6 +179,9 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
     // We can only insert trailers at the end of data, so keep track of this value
     // until then.
     grpc_status_ = grpcStatusFromHeaders(headers);
+
+    // gRPC clients expect that the HTTP status will always be 200.
```

Comment:
I think you should do it only when it is a gRPC response. For regular HTTP responses the status should not be modified.

Also as this is a user visible change it will require a runtime flag protection.

Reply:
Runtime flags are defined here: https://github.com/envoyproxy/envoy/blob/main/source/common/runtime/runtime_features.cc